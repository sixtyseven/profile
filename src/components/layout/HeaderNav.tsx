import React from "react"
import { Link } from "gatsby"
import { Navbar, Nav, NavDropdown } from "react-bootstrap"
import LinkedInSVG from "src/assets/images/glyphicons-social-18-linked-in.svg"
import GithubSVG from "src/assets/images/glyphicons-social-22-github.svg"
import styles from "./HeaderNav.module.scss"
import { WindowLocation } from "@reach/router"

interface IProps {
  siteTitle: string
  linkedInUrl: string
  gitRepoUrl: string
  location: WindowLocation
}

const HeaderNav = ({
  siteTitle,
  linkedInUrl,
  gitRepoUrl,
  location,
}: IProps) => {
  const dropdownActionClass =
    ["/contact/contact_me", "/contact/report_a_bug"].indexOf(
      location.pathname.replace(/\/$/, "")
    ) !== -1
      ? "active"
      : ""

  return (
    <>
      <Navbar collapseOnSelect variant="dark" expand="lg">
        <Navbar.Brand as={Link} to="/">
          {siteTitle}
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto">
            <Nav.Link as={Link} to="/" activeClassName="active">
              Home
            </Nav.Link>
            <Nav.Link as={Link} to="/skills" activeClassName="active">
              Skills
            </Nav.Link>
            <Nav.Link as={Link} to="/user_guide" activeClassName="active">
              User Guide
            </Nav.Link>
            <NavDropdown
              title="Contact"
              id="basic-nav-dropdown"
              className={`${dropdownActionClass}`}
            >
              <NavDropdown.Item
                as={Link}
                to="/contact/contact_me"
                activeClassName="active"
              >
                Contact Me
              </NavDropdown.Item>
              <NavDropdown.Item
                as={Link}
                to="/contact/report_a_bug"
                activeClassName="active"
              >
                Report a bug
              </NavDropdown.Item>
            </NavDropdown>
          </Nav>
          <div className={`pr-md-5 ${styles.contactIcons}`}>
            <a href={`${gitRepoUrl}`} target="_blank">
              <GithubSVG />
            </a>

            <a href={`${linkedInUrl}`} target="_blank">
              <LinkedInSVG />
            </a>
          </div>
        </Navbar.Collapse>
      </Navbar>
    </>
  )
}

export default HeaderNav
