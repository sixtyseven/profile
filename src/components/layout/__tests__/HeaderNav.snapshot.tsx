import React from "react"
import renderer from "react-test-renderer"
import HeaderNav from "src/components/layout/HeaderNav"
import { WindowLocation } from "@reach/router"

describe("HeaderNav", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <HeaderNav
          siteTitle="Default Starter"
          linkedInUrl="linkedInUrl"
          gitRepoUrl="gitRepoUrl"
          location={{ pathname: "/contact/contact_me" } as WindowLocation}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
