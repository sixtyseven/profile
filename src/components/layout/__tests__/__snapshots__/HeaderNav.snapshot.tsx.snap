// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Default Starter
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        Home
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/skills"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        Skills
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/user_guide"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        User Guide
      </a>
      <div
        className="active dropdown nav-item"
        onKeyDown={[Function]}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
          tabIndex={-1}
        >
          Contact
        </a>
        <div
          aria-labelledby={null}
          className="dropdown-menu"
        >
          <a
            className="dropdown-item"
            disabled={false}
            href="/contact/contact_me"
            onClick={[Function]}
          >
            Contact Me
          </a>
          <a
            className="dropdown-item"
            disabled={false}
            href="/contact/report_a_bug"
            onClick={[Function]}
          >
            Report a bug
          </a>
        </div>
      </div>
    </div>
    <div
      className="pr-md-5 contactIcons"
    >
      <a
        href="gitRepoUrl"
        target="_blank"
      >
        <test-file-stub />
      </a>
      <a
        href="linkedInUrl"
        target="_blank"
      >
        <test-file-stub />
      </a>
    </div>
  </div>
</nav>
`;
